exploration/number-selection.md

```diff
@@ -0,0 +1,179 @@
+
+## Proposed Design
+
+Given that we already have a `:number`,
+it makes sense to add a `<matchSignature>` to it with an option
+
+```xml
+<option name="select" values="plural ordinal exact" default="plural" />
+```
+
+The default `plural` value is presumed to be the most common use case,
+and it affords the least bad fallback when used incorrectly:
+Using "plural" for "exact" still selects exactly matching cases,
+whereas using "exact" for "plural" will not select LDML category matches.
+This might not be noticeable in the source language,
+but can cause problems in target locales that the original developer is not considering.
+
+> For example, a naive developer might use a special message for the value `1` without
+> considering other locale's need for a `one` plural:
+>
+> ```
+> .match {$var}
+> [1] {{You have one last chance}}
+> [one] {{You have {$var} chance remaining}} // needed by languages such as Polish or Russian
+> [*] {{You have {$var} chances remaining}}
+> ```
+
+Additional options such as `minimumFractionDigits` and others already supported by `:number`
+should also be supported.
+
+If PR [#532](https://github.com/unicode-org/message-format-wg/pull/532) is accepted,
+also add the following `<alias>` definitions to `<function name="number">`:
+
+```xml
+<alias name="plural" supports="match">
+	<setOption name="select" value="plural"/>
+</alias>
+<alias name="ordinal" supports="match">
+	<setOption name="select" value="ordinal"/>
+</alias>
+```
+
+## Alternatives Considered
+
+### Completely Separate Functions
+
+An alternative approach to this problem could be to leave the `:number` `<matchSignature>` undefined,
+and to define three further functions, each with a `<matchSignature>`:
+
+- `:plural`
+- `:ordinal`
+- `:exact`
+
+which would each need the same set of options as `:number`, except for `type`.
+
+This approach would also mostly work, but it introduces new failure modes:
+
+- If a `:number` is used directly as a selector, this should produce a runtime error.
+- If a `:plural`/`:ordinal`/`:exact` is used as a formatter, this should produce a runtime error.
+- Developers are less encouraged to use the same formatting and selection options.
+
+To expand on the last of these,
+consider this message:
+
+```
+.match {$count :plural minimumFractionDigits=1}
+0 {{You have no apples}}
+1 {{You have exactly one apple}}
+* {{You have {$count :number minimumFractionDigits=1} apples}}
+```
+
+Here, because selection on `:number` is not allowed,
+it's easy to duplicate the options because _some_ annotation is required on the selector.
+It would also be relatively easy to leave out the `minimumFractionDigits=1` option from the selector,
+as it's not required for the English source.
+
+With the proposed design, this message would much more naturally be written as:
+
+```
+.input {$count :number minimumFractionDigits=1}
+.match {$count}
+0 {{You have no apples}}
+1 {{You have exactly one apple}}
+* {{You have {$count} apples}}
```

Comment:
As noted before we should model good behavior. This message should also have a `one` rule

```suggestion
one {{You have {$count} apple}}
* {{You have {$count} apples}}
```